mlir/lib/Target/LLVMIR/DebugImporter.cpp

```diff
@@ -116,6 +116,16 @@ DebugImporter::translateImpl(llvm::DILexicalBlockFile *node) {
                                      node->getDiscriminator());
 }
 
+DIGlobalVariableAttr
+DebugImporter::translateImpl(llvm::DIGlobalVariable *node) {
+  return DIGlobalVariableAttr::get(
+      context, translate(node->getScope()),
+      StringAttr::get(context, node->getName()),
+      StringAttr::get(context, node->getLinkageName()),
```

Comment:
Considering that you always set the `linkageName`, it might be reasonable to not make this an optional parameter.

Reply:
LLVM's own docs showed a case where it was not set (assuming it is correct everywhere)

https://llvm.org/docs/LangRef.html#disubrange

The example there was the only place like that.